javaparser-core/src/main/java/com/github/javaparser/TokenRange.java

```diff
@@ -0,0 +1,38 @@
+package com.github.javaparser;
+
+import static com.github.javaparser.utils.Utils.assertNotNull;
+
+/**
+ * The range of tokens covered by this node.
+ */
+public class TokenRange {
+    public static final TokenRange INVALID = new TokenRange(JavaToken.INVALID, JavaToken.INVALID);
+    
+    private final JavaToken begin;
+    private final JavaToken end;
+
+    public TokenRange(JavaToken begin, JavaToken end) {
+        this.begin = assertNotNull(begin);
+        this.end = assertNotNull(end);
+    }
+
+    public JavaToken getBegin() {
+        return begin;
+    }
+
+    public JavaToken getEnd() {
+        return end;
+    }
+
+    public Range getRange() {
```

Comment:
what about `toRange`? It is more of a conversion than extracting a value. Just a thought, not feeling strongly about this at all